bilby/gw/likelihood.py

```diff
@@ -1526,76 +1529,53 @@ def log_likelihood_ratio(self):
         d_inner_h = 0.
         optimal_snr_squared = 0.
         complex_matched_filter_snr = 0.
+
         self.parameters.update(self.get_sky_frame_parameters())
+
         waveform_ratio = self.compute_waveform_ratio(self.parameters)
 
         for interferometer in self.interferometers:
-            per_detector_snr = self.calculate_snrs_relative_binning(waveform_ratio[interferometer.name], interferometer)
-
+            per_detector_snr = self.calculate_snrs_relative_binning(
+                waveform_ratio[interferometer.name], interferometer)
             d_inner_h += per_detector_snr.d_inner_h
-            optimal_snr_squared += np.real(
-                per_detector_snr.optimal_snr_squared)
+            optimal_snr_squared += np.real(per_detector_snr.optimal_snr_squared)
             complex_matched_filter_snr += per_detector_snr.complex_matched_filter_snr
 
         log_l = np.real(d_inner_h) - optimal_snr_squared / 2
         return float(log_l.real)
 
-    def find_maximum_likelihood_waveform(self, initial_parameter_guess,
-                                         parameter_bounds, iterations=10,
-                                         likelihood_threshold=1):
-        prev_log_likelihood = -np.inf
+    def find_maximum_likelihood_parameters(self, parameter_bounds,
+                                           iterations=1, atol=1e-3, maxiter=100):
```

Comment:
You can just take as input a `optimization_kwargs` parameter (or something like that); it should be a dictionary which would be directly passed into this function.

Reply:
Can you check the changes in https://github.com/adivijaykumar/bilby/pull/30/commits/0f05321fccffd38e0da4248398a41cf3d4f5c827 and let me know if that's the right way to do this?

Reply:
I think it is okay, but there is probably a better way to do things.

Reply:
Check #34

Reply:
Is passing the parameters as **kwargs not a good practice or something?

Reply:
I think it can cause conflicts: `**maximization_kwargs` will pass the whole dictionary converted into options, while `maximization_kwargs=maximization_kwargs` will make a different option for the arguments. Since you don't need `maximization_kwargs` anywhere except in the `differential_evolution` function, it is better if you pass it as `**maximization_kwargs` only in `differential_evolution`.